src/Riok.Mapperly/SyntaxProvider.Roslyn4.4.cs

```diff
@@ -8,6 +9,14 @@ namespace Riok.Mapperly;
 
 internal static class SyntaxProvider
 {
+    public static IncrementalValueProvider<ImmutableArray<Compilation>> GetNestedCompilations(
+        IncrementalGeneratorInitializationContext context
+    ) =>
+        context
+            .MetadataReferencesProvider.Select((metadataReference, _) => (metadataReference as CompilationReference)?.Compilation!)
+            .Where(x => x is not null)
+            .Collect();
```

Comment:
Why the not-null assertion + where `not null`?
Probably simplify to
```csharp
context
            .MetadataReferencesProvider
            .Select((metadataReference, _) => (metadataReference as CompilationReference)?.Compilation)
            .WhereNotNull()
            .Collect();
```
or use the same new `OfType` extension method proposed in the comment for roslyn 4.0.